scanpy/tests/test_preprocessing.py

```diff
@@ -115,6 +115,22 @@ def test_subsample_copy():
     assert sc.pp.subsample(adata, fraction=0.1, copy=True).shape == (20, 10)
 
 
+def test_subsample_copy_backed(tmp_path):
+    A = np.random.rand(200, 10).astype(np.float32)
+    adata_m = AnnData(A.copy())
+    adata_d = AnnData(A.copy())
+    filename = tmp_path / 'test.h5ad'
+    adata_d.filename = filename
+    # This should not throw an error
+    assert sc.pp.subsample(adata_d, n_obs=40, copy=True).shape == (40, 10)
+    assert np.allclose(
```

Comment:
```suggestion
    np.testing.assert_array_equal(
```

* The np.testing functions can give better error messages
* I think these should be equal, not just close